dialogue-core/src/main/metrics/dialogue-core-metrics.yml

```diff
@@ -20,6 +20,10 @@ namespaces:
           Number of requests that are actively running. The `stage` may refer to `running` requests actively
           executing over the wire or `processing` which may be awaiting a client or backing off for a retry.
           Note that running requests are also counted as processing.
+      request.retry:
+        type: counter
```

Comment:
I think a meter makes more sense here, we're generally measuring rate of retires over time rather than the concrete value.